website_sale_custom_filter/models/product.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Camptocamp SA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        res = super()._search_get_detail(website, order, options)
+        # modify base_domain to include selected filters
+        domain = res["base_domain"][0]
+
+        CustomFilterObj = self.env["website.sale.custom.filter"]
+        CustomFilterValueObj = self.env["website.sale.custom.filter.value"]
+
+        if options.get("custom_checkbox_filters", False):
+            for _, val_id in options["custom_checkbox_filters"].items():
+                filter_value = CustomFilterValueObj.browse(val_id)
+                selected_products = filter_value.selected_product_tmpl_ids.ids
+                if selected_products:
+                    domain.append(("id", "in", selected_products))
+        if options.get("custom_value_filters", False):
+            for f_id, val_id in options["custom_value_filters"].items():
+                filter_id = CustomFilterObj.browse(int(f_id))
+                filtering_field = filter_id.numerical_filter_field_id.name
+                min_val = val_id.get("min_value", False)
+                max_val = val_id.get("max_value", False)
+                # build domain
+                if min_val and min_val != val_id["available_min_value"]:
+                    domain.append((filtering_field, ">=", min_val))
+                if max_val and max_val != val_id["available_max_value"]:
+                    domain.append((filtering_field, "<=", max_val))
+        res["base_domain"][0] = domain
```

Comment:
IMO appending elements into the domain is a dangerous way to extend an existing domain.
I would instead use `osv.expression.AND` and `osv.expression.OR` to do it safely.
(I again feel that splitting the function would make it more readable)
```
@api.model
def _search_get_detail(self, website, order, options):
    res = super()._search_get_detail(website, order, options)
    original_domain = res["base_domain"][0]
    new_domain_elements = []

    if options.get("custom_checkbox_filters", False):
        checkbox_filter_domain = self._get_checkbox_filter_domain(options.get("custom_checkbox_filters"))
        if checkbox_filter_domain:
            new_domain_elements.append(checkbox_filter_domain)

    if options.get("custom_value_filters", False):
        value_filter_domain = self._get_value_filter_domain(options.get("custom_value_filters"))
        if value_filter_domain:
            new_domain_elements = .append(value_filter_domain)
    if new_domain_elements
    res["base_domain"][0] = AND(original_domain + new_domain_elements)
    return res

def _get_checkbox_filter_domain(self, option_values):
    subdomains = []
    for _, val_id in options["custom_checkbox_filters"].items():
        filter_value = CustomFilterValueObj.browse(val_id)
        selected_products = filter_value.selected_product_tmpl_ids.ids
        if selected_products:
            subdomains.append([("id", "in", selected_products)])
    return OR(subdomains)

def _get_value_filter_domain(self, option_values):
    subdomains = []
    for f_id, val_id in options["custom_value_filters"].items():
        filter_id = CustomFilterObj.browse(int(f_id))
        filtering_field = filter_id.numerical_filter_field_id.name
        min_val = val_id.get("min_value", False)
        max_val = val_id.get("max_value", False)
        # build domain
        if min_val and min_val != val_id["available_min_value"]:
            subdomains.append([(filtering_field, ">=", min_val)])
        if max_val and max_val != val_id["available_max_value"]:
            subdomains.append([(filtering_field, "<=", max_val)])
    return AND(subdomains)
```